File: js/palindrome.js
export function isPalindrome1(input) {
  for (let i = 0; i < input.length; i++) {
    if (input.charAt(i) != input.charAt(input.length - 1 - i)) {
        return false;
    }
  }
  return true;
}

export function isPalindrome2(input) {
  const reversed = input.toString()
    .split("")
    .reverse()
    .join("");
  return input == reversed;
}

console.log(isPalindrome("abcba"));
console.log(isPalindrome("abc"));

File: java/palindrome.java
class PalindromeExmaple {
    public static boolean isPalindrome1(String input) {
        for (int i = 0; i < input.length(); i++) {
            if (input.charAt(i) != input.charAt(input.length() - 1 - i)) {
                return false;
            }
        }
        return true;
    }
    
    public static void main(String args[]) {
        System.out.println(isPalindrome1("abcba"));
        System.out.println(isPalindrome1("abc"));
    }
}
